<launch>

  <node pkg="nodelet" type="nodelet" name="nodelet_manager" args="manager" output="screen" />

  <node pkg="nodelet" type="nodelet" name="blackfly_nodelet" args="load blackfly/blackfly_nodelet nodelet_manager" clear_params="true" output="screen">

    <!-- Serial Numbers of Cameras -->
    <rosparam param="camera_serial_nums">["21290732"]</rosparam>
    <!-- Names of Cameras (used for topic names, etc) -->
    <rosparam param="camera_names">["cam0"]</rosparam>
    <!-- Paths to Camera info files -->
    <rosparam param="camera_info_paths">["package://blackfly/camera_info/cam_0.info"]</rosparam>

    <!-- Publish Mono Images Flag : True = Mono8, False = BGR8 -->
    <rosparam param="mono_flags">[false]</rosparam>
    <!-- Is the camera triggered? True = Line0, False = FrameRate -->
    <rosparam param="is_triggered_flags">[true]</rosparam>
    <!-- Time in microseconds between when the camera receives a trigger and
    when exposure begins for the image  -->
    <rosparam param="trigger_delays">[13.0]</rosparam>
    <!-- Frame Rate if not triggered-->
    <rosparam param="fps">[20.0]</rosparam>

    <!-- Use Auto Exposure flag -->
    <rosparam param="is_auto_exp_flags">[true]</rosparam>
    <!-- Auto Exposure Bounds if using Auto Exposure (uSecs) -->
    <rosparam param="max_auto_exp">     [10000.0]</rosparam>
    <rosparam param="min_auto_exp">     [50.0]</rosparam>
    <!-- Fixed Exposure if not Auto (uSecs) -->
    <rosparam param="fixed_exp">        [20000.0]</rosparam>

    <!-- Use Auto Gain Flag -->
    <rosparam param="auto_gains">       [true]</rosparam>
    <!-- Gain Bounds if using Auto Gain -->
    <rosparam param="max_gains">        [2.0]</rosparam>
    <rosparam param="min_gains">        [0.5]</rosparam>
    <!-- Fixed Gain if not Auto Gain -->
    <rosparam param="gains">            [1.0]</rosparam>

    <!-- Enable Gamma Correction -->
    <rosparam param="enable_gamma">     [true]</rosparam>
    <!-- Gamma setting if using Gamma Correction -->
    <rosparam param="gammas">           [1.0]</rosparam>
    <!-- Binning Value (decimate image resolution by x) -->
    <rosparam param="binnings">           [1]</rosparam>
    <!-- <rosparam param="binnings">           [1]</rosparam> -->
    <!-- 0 Average / 1 Sum -->
    <rosparam param="binning_mode">       [1]</rosparam>
    <!-- 0 Normal / 1 backlight / 2 Frontlight -->
    <rosparam param="lighting_mode">       [0]</rosparam>
    <!-- 0 Gain / 1 ExposureTime -->
    <rosparam param="auto_exposure_priority">       [0]</rosparam>
    <!-- Enable Exposure Time Compensation -->
    <rosparam param="exp_comp_flags">     [false]</rosparam>
    <!-- Device Link Throughput Limits for Ethernet Cameras -->
    <rosparam param="device_link_throughput_limits">     [40325200]</rosparam>

    <!-- Enable Dynamic Reconfigure -->
    <rosparam param="enable_dyn_reconf">  false</rosparam>
  </node>
</launch>
